doc/API.md

```diff
@@ -405,4 +441,7 @@ context to be propagated to a service via `proxy_pass`.
 This family of variables is used in the implementation of the Datadog nginx
 module.
 
+### `datadog_auth_request_hook`
+This is an implementation detail of the module and should not be used.
```

Comment:
I prefer to mention these "internal-only" variables, because a configuration _can_ use them (though it shouldn't). It's unlikely that somebody would accidentally use it, but I'd rather not hide what is there.

One way to deal with this would be to change the names of these "internal-only" variables so that they all begin with `$datadog_internal_`. Then mention in the docs "the module uses a namespace `$datadog_internal_*` of variables that are not to be used except internally by the module."

For now, let's mention them individually, and then do a bulk sweeping under the rug in a subsequent PR.